backend/amd64/proc.ml

```diff
@@ -646,6 +650,45 @@ let frame_required ~fun_contains_calls ~fun_num_stack_slots =
 let prologue_required ~fun_contains_calls ~fun_num_stack_slots =
   frame_required ~fun_contains_calls ~fun_num_stack_slots
 
+(* returned size includes return address *)
+let frame_size ~stack_offset ~fun_contains_calls ~fun_num_stack_slots =
+  if frame_required ~fun_contains_calls ~fun_num_stack_slots then begin
+    let sz =
+      (stack_offset
+       + 8
+       + 8 * fun_num_stack_slots.(0)
+       + 8 * fun_num_stack_slots.(1)
+       + 16 * fun_num_stack_slots.(2)
+       + (if fp then 8 else 0))
+    in Misc.align sz 16
+  end else
+    stack_offset + 8
+
+type slot_offset =
+  | Bytes_relative_to_stack_pointer of int
+  | Bytes_relative_to_domainstate_pointer of int
+
+let slot_offset loc ~reg_class ~stack_offset ~fun_contains_calls
```

Comment:
It is no longer about the `reg_class`
but about the `stack_class`.
The callers need to be updated.

Reply:
To elaborate a bit, in the post-SIMD
world:
- we still have two _register_
  classes (the same as before, only
  the float class can be used for
  128-bit SIMD values);
- we have a distinct notion of _stack_
  class (0 and 1 correspond to the
  64-bit values -as before- and the
  new 2 class is used to spill 128-bit
  values).

Reply:
Done.